diesel/src/sqlite/expression/collate.rs

```diff
@@ -0,0 +1,101 @@
+use backend::Backend;
+use expression::*;
+use query_builder::*;
+use result::QueryResult;
+
+pub trait Collation {
+    fn get_collation<'a>(&'a self) -> &'a str;
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct Collate<Lhs: Expression, Coll: Collation> {
+    left: Lhs,
+    collation: Coll,
```

Comment:
If we use a associated constant, we do not need to store collation explicitly. So maybe use `std::marker::PhantomData` here?